x/ccv/provider/keeper/proposal.go

```diff
@@ -249,17 +249,34 @@ func (k Keeper) PendingCreateProposalIterator(ctx sdk.Context) sdk.Iterator {
 }
 
 // IteratePendingCreateProposal iterates over the pending proposals to create consumer chain clients in order
-// and creates the consumer client if the spawn time has passed, otherwise it will break out of loop and return.
+// and creates the consumer client if the spawn time has passed.
 func (k Keeper) IteratePendingCreateProposal(ctx sdk.Context) {
+	propsToExecute := k.CreateProposalsToExecute(ctx)
+
+	for _, prop := range propsToExecute {
+		err := k.CreateConsumerClient(ctx, prop.ChainId, prop.InitialHeight, prop.LockUnbondingOnTimeout)
+		if err != nil {
+			panic(fmt.Errorf("consumer client could not be created: %w", err))
+		}
+	}
+}
+
+// CreateProposalsToExecute iterates over the pending proposals and returns an ordered list of proposals to be executed,
+// ie. consumer clients to be created. A prop is included in the returned list (and is deleted from store) if its proposed
+// spawn time has passed.
+//
+// Note: this method is split out from IteratePendingCreateProposal to be easily unit tested.
+func (k Keeper) CreateProposalsToExecute(ctx sdk.Context) []types.CreateConsumerChainProposal {
 
 	iterator := k.PendingCreateProposalIterator(ctx)
 	defer iterator.Close()
 
+	// store the (to be) executed proposals in order
+	propsToExecute := []types.CreateConsumerChainProposal{}
```

Comment:
nit: the assignment to iterator and the check for validity should be together

Reply:
I was suggesting sth like this:
```
	// store the (to be) executed proposals in order
	propsToExecute := []types.CreateConsumerChainProposal{}

	iterator := k.PendingCreateProposalIterator(ctx)
	defer iterator.Close()

	if !iterator.Valid() {
		return propsToExecute
	}
```

Reply:
Ah a lot more simple than I assumed :) sounds good